File: event-schemas/data/csharp/AssetVariable.cs
namespace entities
{
  using System.Collections.Generic;

  public class AssetVariable
  {
    private VariableType type;
    private string key;
    private double value;
    private AssetVariation[] variations;
    private Dictionary<string, dynamic> additionalProperties;

    public VariableType Type 
    {
      get { return type; }
      set { type = value; }
    }

    public string Key 
    {
      get { return key; }
      set { key = value; }
    }

    public double Value 
    {
      get { return value; }
      set { value = value; }
    }

    public AssetVariation[] Variations 
    {
      get { return variations; }
      set { variations = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}

File: event-schemas/src/entities/AssetVariable.ts
import VariableType from './VariableType';
import AssetVariation from './AssetVariation';

class AssetVariable {
  private _type: VariableType;
  private _key: string;
  private _value: number;
  private _variations?: Array<AssetVariation>;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    type: VariableType,
    key: string,
    value: number,
    variations?: Array<AssetVariation>,
  }) {
    this._type = input.type;
    this._key = input.key;
    this._value = input.value;
    this._variations = input.variations;
  }

  get type(): VariableType { return this._type; }
  set type(type: VariableType) { this._type = type; }

  get key(): string { return this._key; }
  set key(key: string) { this._key = key; }

  get value(): number { return this._value; }
  set value(value: number) { this._value = value; }

  get variations(): Array<AssetVariation> | undefined { return this._variations; }
  set variations(variations: Array<AssetVariation> | undefined) { this._variations = variations; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default AssetVariable;
